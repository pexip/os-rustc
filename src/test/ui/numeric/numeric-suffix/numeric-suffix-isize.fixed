// run-rustfix

fn foo<N>(_x: N) {}

fn main() {
    foo::<isize>(42_isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_isize);
    foo::<isize>(42_isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42isize);
    //~^ ERROR mismatched types
    //~| NOTE expected
}
