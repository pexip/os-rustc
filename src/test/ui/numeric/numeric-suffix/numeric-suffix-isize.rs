// run-rustfix

fn foo<N>(_x: N) {}

fn main() {
    foo::<isize>(42_usize);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_u64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_u32);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_u16);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_u8);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_isize);
    foo::<isize>(42_i64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_i32);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_i16);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42_i8);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42.0_f64);
    //~^ ERROR mismatched types
    //~| NOTE expected
    foo::<isize>(42.0_f32);
    //~^ ERROR mismatched types
    //~| NOTE expected
}
