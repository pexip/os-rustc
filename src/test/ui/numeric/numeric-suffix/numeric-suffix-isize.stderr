error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:6:18
   |
LL |     foo::<isize>(42_usize);
   |                  ^^^^^^^^ expected `isize`, found `usize`
   |
help: change the type of the numeric literal from `usize` to `isize`
   |
LL |     foo::<isize>(42_isize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:9:18
   |
LL |     foo::<isize>(42_u64);
   |                  ^^^^^^ expected `isize`, found `u64`
   |
help: change the type of the numeric literal from `u64` to `isize`
   |
LL |     foo::<isize>(42_isize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:12:18
   |
LL |     foo::<isize>(42_u32);
   |                  ^^^^^^ expected `isize`, found `u32`
   |
help: change the type of the numeric literal from `u32` to `isize`
   |
LL |     foo::<isize>(42_isize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:15:18
   |
LL |     foo::<isize>(42_u16);
   |                  ^^^^^^ expected `isize`, found `u16`
   |
help: change the type of the numeric literal from `u16` to `isize`
   |
LL |     foo::<isize>(42_isize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:18:18
   |
LL |     foo::<isize>(42_u8);
   |                  ^^^^^ expected `isize`, found `u8`
   |
help: change the type of the numeric literal from `u8` to `isize`
   |
LL |     foo::<isize>(42_isize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:22:18
   |
LL |     foo::<isize>(42_i64);
   |                  ^^^^^^ expected `isize`, found `i64`
   |
help: change the type of the numeric literal from `i64` to `isize`
   |
LL |     foo::<isize>(42_isize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:25:18
   |
LL |     foo::<isize>(42_i32);
   |                  ^^^^^^ expected `isize`, found `i32`
   |
help: change the type of the numeric literal from `i32` to `isize`
   |
LL |     foo::<isize>(42_isize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:28:18
   |
LL |     foo::<isize>(42_i16);
   |                  ^^^^^^ expected `isize`, found `i16`
   |
help: change the type of the numeric literal from `i16` to `isize`
   |
LL |     foo::<isize>(42_isize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:31:18
   |
LL |     foo::<isize>(42_i8);
   |                  ^^^^^ expected `isize`, found `i8`
   |
help: change the type of the numeric literal from `i8` to `isize`
   |
LL |     foo::<isize>(42_isize);
   |                     ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:34:18
   |
LL |     foo::<isize>(42.0_f64);
   |                  ^^^^^^^^ expected `isize`, found `f64`
   |
help: change the type of the numeric literal from `f64` to `isize`
   |
LL |     foo::<isize>(42isize);
   |                    ~~~~~

error[E0308]: mismatched types
  --> $DIR/numeric-suffix-isize.rs:37:18
   |
LL |     foo::<isize>(42.0_f32);
   |                  ^^^^^^^^ expected `isize`, found `f32`
   |
help: change the type of the numeric literal from `f32` to `isize`
   |
LL |     foo::<isize>(42isize);
   |                    ~~~~~

error: aborting due to 11 previous errors

For more information about this error, try `rustc --explain E0308`.
